python/ray/autoscaler/updater.py

```diff
@@ -335,7 +358,22 @@ def run(self,
             timeout=timeout,
             exit_on_fail=exit_on_fail,
             port_forward=None,
-            with_output=False)
+            with_output=False,
+            ssh_options=ssh_options)
+
+    def check_docker_installed(self):
+        install_commands = [
+            "curl -fsSL https://get.docker.com -o get-docker.sh",
+            "sudo sh get-docker.sh", "sudo usermod -aG docker $USER",
+            "sudo systemctl restart docker -f"
+        ]
+        try:
+            self.ssh_command_runner.run("command -v docker")
+            return
+        except Exception:
+            logger.error(
+                "Docker not installed, please add the following commands to "
+                "initialization_commands:\n" + "\n".join(install_commands))
```

Comment:
Really nice! We should be doing this more.

```suggestion
                "Docker not installed. You can install Docker by adding the following commands to "
                "initialization_commands:\n" + "\n".join(install_commands))
```